precompile/stateful_precompile_config.go

```diff
@@ -33,9 +45,23 @@ type StatefulPrecompileConfig interface {
 	// provides the config the ability to set its initial state and should only modify the state within
 	// its own address space.
 	Configure(ChainConfig, StateDB, BlockContext)
+
 	// Contract returns a thread-safe singleton that can be used as the StatefulPrecompiledContract when
 	// this config is enabled.
 	Contract() StatefulPrecompiledContract
+
+	// Predicate returns an optional function which is called a predicate on every transaction's access list
+	// for any access list tuple whose address matches the address of the precompile.
+	// This allows the precompile to enforce a predicate on the transaction itself for it to be considered valid
+	// to be included in a block.
+	Predicate() PredicateFunc
+
+	// OnAccept returns an optional function which is called when a block gets accepted on each log whose
+	// address matches the address of the precompile.
+	// This can be used to perform precompile specific logic on acceptance, so that the precompile can emit
+	// events and perform logic on those events only after the block has been accepted.
+	OnAccept() OnAcceptFunc
```

Comment:
where in the code do we call `OnAccept()`? Also I'm thinking how we'd have access to the warp backend to call `AddMessage()` within `OnAccept()` given the current parameters.